src/server/server.ts

```diff
@@ -107,8 +107,19 @@ const startProxyServer = (): void => {
       return res.end(`Error: failed to create proxy ${req.headers.host}`)
     }
   })
-  server.listen(443)
 
+  server.on('upgrade', function(req, socket, head) {
+    const mappings = getMappings()
+    const { ip, port }: ProxyMapping =
+      mappings.find(({ subDomain, domain }) => {
+        const prefix = subDomain ? `${subDomain}.` : ''
+        return `${prefix}${domain}` === req.headers.host
+      }) || {}
+    if (port)
+      return proxy.ws(req, socket, { target: `http://127.0.0.1:${port}` })
```

Comment:
don't hardcode the ip address, you already got the ip address from the variable.